cli/azd/extensions/azure.foundry.ai.agents/internal/cmd/init.go

```diff
@@ -1237,25 +1289,25 @@ func (a *InitAction) updateEnvironment(ctx context.Context, agentManifest *agent
 	envName := envResponse.Environment.Name
 
 	// Set environment variables based on agent kind
-	switch agentManifest.Agent.Kind {
-	case "hosted":
+	switch agentDef.Kind {
+	case agent_yaml.AgentKindPrompt:
+		agentDef := agentManifest.Template.(agent_yaml.PromptAgent)
+		if err := a.setEnvVar(ctx, envName, "AZURE_AI_FOUNDRY_MODEL_NAME", agentDef.Model.Id); err != nil {
+			return err
+		}
+	case agent_yaml.AgentKindHosted:
 		// Set environment variables for hosted agents
 		if err := a.setEnvVar(ctx, envName, "ENABLE_HOSTED_AGENTS", "true"); err != nil {
 			return err
 		}
-	case "container":
+	case agent_yaml.AgentKindYamlContainerApp:
 		// Set environment variables for foundry agents
 		if err := a.setEnvVar(ctx, envName, "ENABLE_CONTAINER_AGENTS", "true"); err != nil {
 			return err
 		}
 	}
```

Comment:
I know some of this is going away, but are we still going to be setting `ENABLE_HOSTED_AGENTS` and `ENABLE_CONTAINER_AGENTS`?

When testing I found that when provisioning a HOBO agent, azd would prompt for the value for `enableContainerAgents` (`ENABLE_CONTAINER_AGENTS`). Are the two mutually exclusive? 

Wondering if it makes sense to set the other variable to `false` so that azd doesn't prompt the user.